File: Go/leetcode/14.go
package Leetcode

// LongestCommonPrefix
// 编写一个函数来查找字符串数组中的最长公共前缀。
// 如果不存在公共前缀，返回空字符串 ""
func LongestCommonPrefix(strs []string) string {
	// 是否为空
	if len(strs) == 0 {
		return ""
	}
	// 最短字符串
	minStr := strs[0]
	for _, str := range strs {
		if len(str) < len(minStr) {
			minStr = str
		}
	}
	// 最长公共前缀
	for i := 0; i < len(strs); i++ {
		for j := 0; j < len(minStr); j++ {
			if strs[i][j] != minStr[j] {
				minStr = minStr[:j]
				break
			}
		}
	}
	return minStr
}

File: Python/leetcode/14.py
from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        # 判断是否为空
        if not strs:
            return ""
        # 找出最小长度的字符串
        minStr = strs[0]
        for min_str in strs:
            if len(min_str) < len(minStr):
                minStr = min_str

        # 最长公共前缀
        for i in range(len(minStr)):
            for j in strs:
                if j[i] != minStr[i]:
                    return minStr[:i]
        return minStr

if __name__ == '__main__':
    s = Solution()
    print(s.longestCommonPrefix(["flower", "flow", "flight"]))
    print(s.longestCommonPrefix(["dog", "racecar", "car"]))